main.tf

```diff
@@ -88,6 +88,7 @@ resource "ibm_resource_key" "resource_key" {
 locals {
   cos_instance_id          = var.create_cos_instance ? ibm_resource_instance.cos_instance[0].id : var.existing_cos_instance_id
   cos_instance_guid        = var.create_cos_instance ? ibm_resource_instance.cos_instance[0].guid : element(split(":", var.existing_cos_instance_id), length(split(":", var.existing_cos_instance_id)) - 3)
+  cos_instance_name        = var.create_cos_instance ? var.cos_instance_name : null
```

Comment:
its a bad practise to get an output from the input variable. Please get the value from the `ibm_resource_instance` resource